src/mocking.ts

```diff
@@ -133,14 +133,12 @@ function mockStructure<T extends StructureConstant>(structureType: T, mockedProp
   const count = (structureCounters[structureType] ?? 0) + 1;
 
   structureCounters[structureType] = count;
+  const id = `${structureType}${count}` as Id<ConcreteStructure<T>>;
```

Comment:
Performing this calculation one and locking down the type removed the recursive typing.